### DISCUSSION TITLE
How to contribute to "ugit"?

### DISCUSSION THREAD

**Discussion by yaten2302** [Bhupesh-V/ugit#62] [Q&A]
Hi @Bhupesh-V,
I want to contribute to "ugit", but I'm new to shell scripting, I don't have much knowledge about bash and shell, can you please guide me that from where should I start learning?

**Comment by Bhupesh-V** [comment]
> Hi @Bhupesh-V, I want to contribute to "ugit", but I'm new to shell scripting, I don't have much knowledge about bash and shell, can you please guide me that from where should I start learning?

Hey @yaten2302 Thanks for reaching out.

Not much in-depth knowledge of bash is required. I usually recommend folks this [resource](https://learnxinyminutes.com/docs/bash/) to get a quick overview of shell scripting. Most of the bash & shell scripting principles used in ugit can be found in the resource. Some other resources:
- [devhint's cheatsheet](https://devhints.io/bash)

The rest is just writing few scripts yourself and executing them.

Hope that helps. LMK if you face any isssues/doubts

**Reply by yaten2302** [reply]
Hii, for running bash scripts, do we require linux? or it can be done on VSCode as well?
